src/webview/css/css.js

```diff
@@ -5,488 +5,16 @@ import cssPygments from './cssPygments';
-  float: left;
-  width: 2.2rem;
-  height: 2.2rem;
-}
-.twitter-image {
-  text-align: center;
-  margin: 8px auto;
-}
-.message-tags {
-  text-align: right;
-  margin: 4px 0;
-  font-size: 0.7rem;
-}
-.message-tag {
-  padding: 4px 4px;
-  margin-left: 4px;
-  border-radius: 3px;
-  color: hsla(0, 0%, 50%, 0.75);
-  background: hsla(0, 0%, 0%, 0.1);
-}
-.reaction-list {
-  margin: 8px 0;
-}
-.reaction {
-  color: hsl(0, 0%, 50%);
-  display: inline-block;
-  padding: 5px 6.5px;
-  border-radius: 3px;
-  border: 1px solid hsla(0, 0%, 50%, 0.75);
-  line-height: 1rem;
-  height: 1rem;
-  margin: 4px 8px 4px 0;
-}
-.reaction img {
-  pointer-events: none;
-  cursor: default;
-  max-height: 1rem;
-  max-width: 1rem;
-  vertical-align: top;
-}
-.self-voted {
-  color: ${BRAND_COLOR};
-  border: 1px solid ${BRAND_COLOR};
-  background: hsla(177.1, 69.7%, 46.7%, 0.1);
-}
-.hidden {
-  display: none;
-}
-.emoji {
-  display: inline-block;
-  height: 1.25rem;
-  width: 1.25rem;
-  white-space: nowrap;
-  color: transparent;
-  vertical-align: text-top;
-}
-.emoji:before {
-  color: white;
-}
-.widget {
-  display: flex;
-  flex-direction: column;
-  align-items: center;
-  justify-content: center;
-  padding: 1rem;
-  background: hsla(0, 0%, 50%, 0.1);
-  border: 1px dashed hsla(0, 0%, 50%, 0.5);
-  border-radius: 0.5rem;
-}
-#typing {
-  display: flex;
-  padding-left: 16px;
-}
-#typing .content {
-  padding-top: 8px;
-  padding-bottom: 16px;
-}
-#typing span {
-  display: inline-block;
-  background-color: hsl(253, 3%, 72%);
-  width: 0.5rem;
-  height: 0.5rem;
-  border-radius: 100%;
-  margin-right: 5px;
-  animation: bob 2s infinite;
-}
-#typing span:nth-child(2) {
-  animation-delay: 0.15s;
-}
-#typing span:nth-child(3) {
-  animation-delay: 0.3s;
-  margin-right: 0;
-}
-@keyframes bob {
-  10% {
-    transform: translateY(-10px);
-    background-color: hsl(253, 3%, 63%);
-  }
-  50% {
-    transform: translateY(0);
-    background-color: hsl(253, 3%, 72%);
-  }
-}
-#message-loading {
-  position: fixed;
-  width: 100%;
-  opacity: 0.25;
-}
-#js-error-detailed {
-  position: fixed;
-  width: 100%;
-  background: red;
-  color: white;
-  font-size: 0.7rem;
-}
-#js-error-plain, #js-error-plain-dummy {
-  z-index: 1000;
-  width: 100%;
-  background: red;
-  color: white;
-  padding: 4px;
-  text-align: center;
-}
-#js-error-plain {
-  position: fixed;
-}
-#js-error-plain-dummy {
-  visibility: hidden;
-}
-#scroll-bottom {
-  position: fixed;
-  z-index: 200;
-  right: 5px;
-  bottom: 15px;
-}
-#scroll-bottom a {
-  display: block;
-  width: 32px;
-  height: 32px;
-  border-radius: 50%;
-  background: hsla(170, 48%, 54%, 0.5);
-}
-#scroll-bottom .text {
-  clip: rect(0 0 0 0);
-  overflow: hidden;
-  position: absolute;
-  height: 1px;
-  width: 1px;
-}
-#scroll-bottom svg {
-  width: 32px;
-  height: 32px;
-  fill: hsla(0, 0%, 100%, 0.75);
-}
-`;
+/** Common CSS properties. */
+const cssVars = `
+:root {
+  --brand-color: ${BRAND_COLOR}
```

Comment:
Per MDN: Chrome 49, Safari 9.1. That puts us in the safe zone for Safari, but the danger zone for Android. I've confirmed that it does _not_ work on the stock KitKat image. It won't work on stock Lollipop, either (Chrome 37), but it will work on a vaguely-updated Lollipop.

... also, it turns out that the code to absolutize relative URLs in a previous commit needs a couple of KitKat polyfills. :sweat:

It might be better to go ahead and do this the "hard" way; it shouldn't actually be that hard. (And either way, despite the branch name, this commit and the next needn't block the rest of the PR.)

Reply:
Rereading myself what we say there, I think it's acceptable as long as this works in Chrome 58, which comes on a stock Oreo image. (The WebView's browser is updated as an APK, and empirically is usually recent even when the OS is old.) And even core functionality can break on older than Chrome 44, found in stock Marshmallow.

I'll start a chat discussion about our Chrome version support, because it's complicated and I don't think we've really thought about it in a year.

Reply:
That discussion here:
https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/Chrome.20versions/near/790982